scipy/stats/tests/test_distributions.py

```diff
@@ -371,6 +371,204 @@ def test_pdf_zero(self):
+        # x <- seq(-10, 10, length.out = 10)
+        # GeneralizedHyperbolic::dghyp(
+        #    x = x, lambda = 2, alpha = 2, beta = 1, delta = 1.5, mu = 0.5
+        # )
+        vals_R = np.array([
+            2.94895678275316e-13, 1.75746848647696e-10, 9.48149804073045e-08,
+            4.17862521692026e-05, 0.0103947630463822, 0.240864958986839,
+            0.162833527161649, 0.0374609592899472, 0.00634894847327781,
+            0.000941920705790324
+            ])
+
+        lmbda = 2
+        alpha = 2
+        beta = 1
+        delta = 1.5
+        mu = 0.5
+
+        alpha_hat = alpha * delta
+        beta_hat = beta * delta
+
+        x = np.linspace(-10, 10, 10)
+        gh = stats.genhyperbolic(
+            p=lmbda,
+            a=alpha_hat,
+            b=beta_hat,
+            loc=mu,
+            scale=delta
+            )
+
+        assert_allclose(gh.pdf(x), vals_R, atol=1e-9)
+
+    def test_cdf_r(self):
+        # test against R package GeneralizedHyperbolic
+        # x <- seq(0, 1, length.out = 10)
+        # GeneralizedHyperbolic::pghyp(
+        #   x = x, lambda = 2, alpha = 2, beta = 1, delta = 1.5, mu = 0.5
+        # )
+        vals_R = np.array([
+            1.01881590921421e-13, 6.13697274983578e-11, 3.37504977637992e-08,
+            1.55258698166181e-05, 0.00447005453832497, 0.228935323956347,
+            0.755759458895243, 0.953061062884484, 0.992598013917513,
+            0.998942646586662
+            ])
+
+        lmbda = 2
+        alpha = 2
+        beta = 1
+        delta = 1.5
+        mu = 0.5
+
+        alpha_hat = alpha * delta
+        beta_hat = beta * delta
+
+        x = np.linspace(-10, 10, 10)
+        gh = stats.genhyperbolic(
+            p=lmbda,
+            a=alpha_hat,
+            b=beta_hat,
+            loc=mu,
+            scale=delta
+            )
+
+        assert_allclose(gh.cdf(x), vals_R, atol=1e-9, rtol=1e-5)
+
+    def test_moments_r(self):
+        # test against R package GeneralizedHyperbolic
+        # sapply(1:4,
+        #    function(x) GeneralizedHyperbolic::ghypMom(
+        #        order = x, lambda = 2, alpha = 2,
+        #        beta = 1, delta = 1.5, mu = 0.5,
+        #        momType = 'central')
+        # )
+
+        vals_R = np.array([
+            -4.44089209850063e-16, 2.86421978532571,
+            4.24895879539469, 37.6380875493332
+            ])
+
+        lmbda = 2
+        alpha = 2
+        beta = 1
+        delta = 1.5
+        mu = 0.5
+
+        alpha_hat = alpha * delta
+        beta_hat = beta * delta
+
+        mvsk = stats.genhyperbolic(
+            p=lmbda,
+            a=alpha_hat,
+            b=beta_hat,
+            loc=mu,
+            scale=delta
+            ).stats('mvsk')
+
+        mvsk = list(mvsk)
+
+        # center and de-standardize moments to match R results
+        mvsk[0] -= mvsk[0]
+        mvsk[2] = mvsk[2] * np.float_power(mvsk[1], 3 / 2)
+        mvsk[3] = (mvsk[3] + 3) * np.float_power(mvsk[1], 2)
```

Comment:
This works, but maybe it could be a bit more direct if you use the raw moments from R.
```python3
vals_R = [2.368483669481150, 8.473934677924596,
          37.887050271006586, 205.766085114849631]
vals_us = [stats.genhyperbolic(p=lmbda, a=alpha_hat, b=beta_hat,
                               loc=mu, scale=delta).moment(i)
           for i in range(1, 5)]
```

Reply:
Agree will change that